File: es70x/src/main/java/tech/beshu/ror/es/RorNotReadyResponse.java
/*
 *    This file is part of ReadonlyREST.
 *
 *    ReadonlyREST is free software: you can redistribute it and/or modify
 *    it under the terms of the GNU General Public License as published by
 *    the Free Software Foundation, either version 3 of the License, or
 *    (at your option) any later version.
 *
 *    ReadonlyREST is distributed in the hope that it will be useful,
 *    but WITHOUT ANY WARRANTY; without even the implied warranty of
 *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *    GNU General Public License for more details.
 *
 *    You should have received a copy of the GNU General Public License
 *    along with ReadonlyREST.  If not, see http://www.gnu.org/licenses/
 */
package tech.beshu.ror.es;

import org.elasticsearch.ElasticsearchStatusException;
import org.elasticsearch.common.xcontent.XContentBuilder;
import org.elasticsearch.rest.RestStatus;

import java.io.IOException;

public class RorNotReadyResponse extends ElasticsearchStatusException {

  public RorNotReadyResponse() {
    super("ReadonlyREST is not ready", RestStatus.SERVICE_UNAVAILABLE);
  }

  @Override
  public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
    builder.field("reason", "Waiting for ReadonlyREST start");
    return builder;
  }
}

File: es74x/src/main/scala/tech/beshu/ror/es/request/RorNotReadyResponse.scala
/*
 *    This file is part of ReadonlyREST.
 *
 *    ReadonlyREST is free software: you can redistribute it and/or modify
 *    it under the terms of the GNU General Public License as published by
 *    the Free Software Foundation, either version 3 of the License, or
 *    (at your option) any later version.
 *
 *    ReadonlyREST is distributed in the hope that it will be useful,
 *    but WITHOUT ANY WARRANTY; without even the implied warranty of
 *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *    GNU General Public License for more details.
 *
 *    You should have received a copy of the GNU General Public License
 *    along with ReadonlyREST.  If not, see http://www.gnu.org/licenses/
 */
package tech.beshu.ror.es.request

import org.elasticsearch.common.xcontent.XContentBuilder
import org.elasticsearch.rest.{BytesRestResponse, RestChannel, RestStatus}
import tech.beshu.ror.es.utils.ErrorContentBuilderHelper.createErrorResponse

class RorNotReadyResponse private (status: RestStatus, builder: XContentBuilder)
  extends BytesRestResponse(status, builder)

object RorNotReadyResponse {
  def create(channel: RestChannel): RorNotReadyResponse = new RorNotReadyResponse(
    RestStatus.SERVICE_UNAVAILABLE,
    createErrorResponse(channel, RestStatus.SERVICE_UNAVAILABLE, RorNotReadyResponse.addRootCause)
  )

  private def addRootCause(builder: XContentBuilder): Unit = {
    builder.field("reason", "Waiting for ReadonlyREST start")
  }
}
